### PR TITLE
TDS2-2069 Session updates

### PR DESCRIPTION
**Pull request by joey-griffin-cko**
### Jira link
-  [TDS2-2069](https://checkout.atlassian.net/browse/TDS2-2069)

Various updates to the sessions API based on review meeting.

### CODE REVIEW

File: spec/components/schemas/Sessions/CreateSessionLinks.yaml [hunk 1]
```diff
@@ -9,4 +9,13 @@ properties:
     properties:
       href:
         type: string
-        example: http://3ds2.checkout.com/interceptor/sid_y3oqhf46pyzuxjbcn2giaqnb44
\ No newline at end of file
+        maxLength: 100
+        example: http://3ds2.checkout.com/interceptor/sid_y3oqhf46pyzuxjbcn2giaqnb44
+  collect_channel_data:
```
Comment (helder-goncalves-cko) [inline spec/components/schemas/Sessions/CreateSessionLinks.yaml]: Does this link even exist?

File: spec/components/schemas/Sessions/DsPublicKey.yaml [hunk 2]
```diff
@@ -3,10 +3,9 @@ description: Public key specific to a DS for encrypting device data
 properties:
   public_key: 
     type: string
+    maxLength: 1024
     example: abc
-  directory_server_id:
```
Comment (helder-goncalves-cko) [inline spec/components/schemas/Sessions/DsPublicKey.yaml]: Why don't we have the ds_id?

File: spec/components/schemas/Sessions/SessionAddress.yaml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    description: The country subdivision code defined in ISO 3166-2
+    minLength: 3
+    maxLength: 3
+    example: ENG
+  zip:
+    type: string
+    description: The address zip/postal code
+    maxLength: 16
+    example: W1T 4TJ
+  country:
```
Comment (helder-goncalves-cko) [inline spec/components/schemas/Sessions/SessionAddress.yaml]: Can you validate if we use `ISO-2` or `ISO-3` elsewhere? We should keep it consistent

File: spec/components/schemas/Sessions/DsPublicKey.yaml (hunk 2 above)
Comment (joey-griffin-cko) [reply spec/components/schemas/Sessions/DsPublicKey.yaml]: Decided to remove it as that's not a value we store right now. We need to properly plan what to do with this endpoint.

File: spec/components/schemas/Sessions/SessionAddress.yaml (hunk 3 above)
Comment (joey-griffin-cko) [reply spec/components/schemas/Sessions/SessionAddress.yaml]: We should be using iso 2 here

File: spec/components/schemas/Sessions/DsPublicKey.yaml (hunk 2 above)
Comment (helder-goncalves-cko) [reply spec/components/schemas/Sessions/DsPublicKey.yaml]: But it's required for the mobile sdks for encryption right?

Comment (michael-giorgaras-cko) [reply spec/components/schemas/Sessions/DsPublicKey.yaml]: Isn't this just our internal id? All we need is to make sure we use the same DS in the client and the server. Which can be controlled by the app serving the certificates

File: spec/components/schemas/Sessions/CardSource.yaml [hunk 4]
```diff
@@ -13,6 +13,8 @@ allOf:
         type: string
         description: The card number (without separators)
         example: "4543474002249996"
+        minLength: 13
```
Comment (michael-giorgaras-cko) [inline spec/components/schemas/Sessions/CardSource.yaml]: I think Maestro uses 19 digit PANs so we better use that for future-proofing?

File: spec/components/schemas/Sessions/ChannelData.yaml [hunk 5]
```diff
@@ -9,5 +9,18 @@ discriminator:
...
+  accept_header: "Accept:  *.*, q=0.1"
+  java_enabled: true
+  language: "FR-fr"
+  color_depth: "16"
+  screen_height: "1080"
+  screen_width: "1920"
+  timezone: "60"
+  user_agent: "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/69.0.3497.100 Safari/537.36"
+  three_ds_method_completion: "Y"
+  ip_address: "1.12.123.255"
```
Comment (michael-giorgaras-cko) [inline spec/components/schemas/Sessions/ChannelData.yaml]: 👍

File: spec/components/schemas/Sessions/CreateSessionLinks.yaml (hunk 1 above)
Comment (michael-giorgaras-cko) [inline spec/components/schemas/Sessions/CreateSessionLinks.yaml]: Are we missing `self` & `issuer_fingerprint`. Also, if we have `collect_channel_data` we can add `complete` perhaps?

File: spec/components/schemas/Sessions/NetworkTokenSource.yaml [hunk 6]
```diff
@@ -31,11 +33,13 @@ allOf:
       name:
         type: string
         description: The customer's name
+        minLength: 2
+        maxLength: 45
```
Comment (michael-giorgaras-cko) [inline spec/components/schemas/Sessions/NetworkTokenSource.yaml]: is 45 coming from EMVco spec? seems a bit low

File: spec/components/schemas/Sessions/DsPublicKey.yaml (hunk 2 above)
Comment (helder-goncalves-cko) [reply spec/components/schemas/Sessions/DsPublicKey.yaml]: It needs to be confirmed if these ids are CKO specific or provided by the scheme.

File: spec/components/schemas/Sessions/CreateSessionLinks.yaml (hunk 1 above)
Comment (helder-goncalves-cko) [reply spec/components/schemas/Sessions/CreateSessionLinks.yaml]: We include those, but we should only display these links when in the correct statuses.

**Comment by michael-giorgaras-cko** [comment]
Also, can we please change `3ds_method_completion` to `three_ds_method_completion`? Or should we change it in the API instead to patch the spec?

File: spec/components/schemas/Sessions/CreateSessionLinks.yaml (hunk 1 above)
Comment (michael-giorgaras-cko) [reply spec/components/schemas/Sessions/CreateSessionLinks.yaml]: Yes like issuer fingerprint

**Comment by michael-giorgaras-cko** [comment]
Also the phone object documents `cc` and `subscriber` as the fields. Let's have `country_code`  and `number` instead please. EMVco field names suck

File: spec/components/schemas/Sessions/Acs.yaml [hunk 7]
```diff
@@ -20,12 +20,14 @@ properties:
   url:
     description: Fully qualified URL of the ACS to be used for the challenge.
     type: string
+    maxLength: 2048
```
[the remark this answers is not in the document]
Comment (joey-griffin-cko) [reply spec/components/schemas/Sessions/Acs.yaml]: Yeah, it's specified in the emvco spec.

File: spec/components/schemas/Sessions/CreateSessionLinks.yaml (hunk 1 above)
Comment (joey-griffin-cko) [reply spec/components/schemas/Sessions/CreateSessionLinks.yaml]: Yep, included are the other links, just not visible in this particular file but it's there in the GetSessionLinks (which merges with this)

File: spec/components/schemas/Sessions/NetworkTokenSource.yaml (hunk 6 above)
Comment (joey-griffin-cko) [reply spec/components/schemas/Sessions/NetworkTokenSource.yaml]: Yep, that's from the spec again.

File: spec/components/schemas/Sessions/DsPublicKey.yaml (hunk 2 above)
Comment (joey-griffin-cko) [reply spec/components/schemas/Sessions/DsPublicKey.yaml]: I'm adding it back in as it's needed as part of the device data encryption process.

File: spec/components/schemas/Sessions/NetworkTokenSource.yaml [hunk 8]
```diff
@@ -15,6 +15,8 @@ allOf:
         type: string
         description: The network token PAN
         example: "4543474002249996"
+        minLength: 9
+        maxLength: 16
```
Comment (nick-kompas-cko) [inline spec/components/schemas/Sessions/NetworkTokenSource.yaml]: this is wrong. It should be from minLength: 9 to maxLength:19

**Pull request merged**